### PR TITLE
cleanup: delete proto struct filter config dead code

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Remove de-activated code using structs for filter configs.

Demonstrate failing unit tests for https://github.com/istio/istio/issues/17461#issuecomment-536150363. The reason is that structs and any are marshaled differently for intermediate embedded message fields. This has no impact on the actual semantics AFAICT.

/assign @howardjohn

### CODE REVIEW

**Comment by kyessenov** [comment]
Lint failing for mysterious reasons...

File: pilot/pkg/networking/util/util.go [hunk 1]
```diff
@@ -291,11 +291,6 @@ func IsIstioVersionGE13(node *model.Proxy) bool {
 		node.IstioVersion.Compare(&model.IstioVersion{Major: 1, Minor: 3, Patch: -1}) >= 0
 }
 
-// IsXDSMarshalingToAnyEnabled controls whether "marshaling to Any" feature is enabled.
-func IsXDSMarshalingToAnyEnabled(node *model.Proxy) bool {
-	return !features.DisableXDSMarshalingToAny
```
Comment (howardjohn) [inline pilot/pkg/networking/util/util.go]: Should also delete `features.DisableXDSMarshalingToAny` if we are making it do nothing?

Comment (kyessenov) [reply pilot/pkg/networking/util/util.go]: Would it break some command line flag? If so, we cannot easily remove it.

Comment (howardjohn) [reply pilot/pkg/networking/util/util.go]: No its just an environment variable. If we keep it around its confusing and actually shows up in our docs so its confusing to users as well

Comment (kyessenov) [reply pilot/pkg/networking/util/util.go]: OK, removed.

**Comment by rshriram** [comment]
can you also cleanup the cruft in v1alpha3/envoyfilter in the networkfilter merge or http filter merge? it checks if the filter is in struct or any form and does some transformations

**Comment by kyessenov** [comment]
@rshriram Can we leave that for later? We still have struct configs that were not guarded (e.g. HCM). Changing that is a higher-risk, and I don't want to couple them in case we need to revert.

**Comment by kyessenov** [comment]
@JimmyCYJ can you please take a look?

**Comment by kyessenov** [comment]
@istio/wg-security-maintainers please take a look. Struct support is being deprecated once again in the upstream envoy.

**Comment by geeknoid** [comment]
To fix the gencheck job, you should either upgrade your compiler to Go 1.13 or at the very least not include go.mod in your PR.

**Comment by kyessenov** [comment]
@istio/wg-security-maintainers please take a look. I'll merge master but this is ready.

File: pilot/pkg/security/authz/builder/builder_test.go [hunk 2]
```diff
@@ -189,63 +181,57 @@ func TestBuilder_BuildTCPFilter(t *testing.T) {
 				policy.SimpleRole("role-1", "a", "foo"),
 				policy.SimpleBinding("binding-1", "a", "role-1"),
 			},
-			wantRules:            true,
+			wantRules:            false,
```
Comment (yangminzhu) [inline pilot/pkg/security/authz/builder/builder_test.go]: Why are these changed from `true` to `false` in the unit tests?

Comment (kyessenov) [reply pilot/pkg/security/authz/builder/builder_test.go]: This is because {a: {b: []}} is marshalled differently between struct roundtripping and directly bytes. What does this test actually test? I couldn't tell and I've been asking around.

Comment (yangminzhu) [reply pilot/pkg/security/authz/builder/builder_test.go]: This is testing the [policies_](https://github.com/envoyproxy/envoy/blob/4825facf2451b234af737b5af8c792c31bca5d41/source/extensions/filters/common/rbac/engine_impl.cc#L34) with empty rules.

Does `Any` changes the semantics in the unmarshalled protobuf in Envoy?

Comment (kyessenov) [reply pilot/pkg/security/authz/builder/builder_test.go]: I don't know - that's why I need you to double-check :) Envoy is removing struct support in the next version, so we should stop generating the structs if we want to be compatible with the latest envoy.

Comment (howardjohn) [reply pilot/pkg/security/authz/builder/builder_test.go]: Any follow up on this? It would be nice to just user Any in master

**Comment by howardjohn** [comment]
Any updates on the failing test?

**Comment by yangminzhu** [comment]
The unit test fail is mostly a false alarm as the e2e tests are all passing, I wanted to test the PR manually but is a little busy with the release-1.4 recently. Is this needed for 1.4? I thought it's just a code refactor.

**Comment by howardjohn** [comment]
Not for 1.4, but Envoy just deprecated struct so we should get rid of it
soon

**Comment by howardjohn** [comment]
@kyessenov are you still working on this?

**Comment by kyessenov** [comment]
This is hopelessly out-of-sync. Closing.
